src/pages/index.js

```diff
@@ -67,21 +99,89 @@ const previewModalCaption = previewModal.querySelector(".modal__caption");
 const previewModalCloseBtn = previewModal.querySelector(
   ".modal__close-button_preview"
 );
+const deleteModal = document.querySelector("#delete-card-modal");
+const deleteForm = deleteModal.querySelector(".delete__card");
+const deleteCancel = deleteModal.querySelector(".modal__cancel");
+const deleteClose = deleteModal.querySelector(".modal__close-button-delete");
 
 const cardTemplate = document.querySelector("#card-template");
 const cardsList = document.querySelector(".cards__list");
 
-function handleNewPostSubmit(evt) {
+let selectedCard;
+let selectedCardId;
+
+function handleAvatarSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: cardCaptionInput.value,
-    link: cardLinkInput.value,
-  };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editAvatarInfo(avatarValue.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
   evt.target.reset();
-  disableButton(cardSubmitBtn, settings);
-  closeModal(newPostModal);
+  closeModal(editAvatarModal);
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**